### ISSUE TITLE
3.3 functional but graphs and logs not working

### ISSUE THREAD

**Issue by tehniemer** [diginc/docker-pi-hole#236]
I updated to 3.3 and now graphs and query logs are empty. However, pi-hole is functioning as expected.

Screenshot the morning after I upgraded
![image 001](https://user-images.githubusercontent.com/11021632/36632186-04d998ce-1949-11e8-84a7-0a45afabc3f3.png)

Screenshot after starting fresh
![image 002](https://user-images.githubusercontent.com/11021632/36632192-162f7b52-1949-11e8-92d6-cd76641f4546.png)

debug log:
[pihole_debug-sanitized.log.log](https://github.com/diginc/docker-pi-hole/files/1754585/pihole_debug-sanitized.log.log)

**Comment by diginc** [comment]
Does http://<your pi hole domain>/admin/api.php?overTimeData10mins
Return some sort of error?  I'm thinking this crops up whenever you have old data, I'm not sure if I'm missing some sort of data transformation process the pi-hole team runs during a version upgrade that could fix this, I'll ping them and see.

I've had to flush my old data (or just delete volume data) to get graphs working after an upgrade.

A while back I disabled the persistent data volumes history because I'm recreating the container to test it a lot and it works more consistently.  I'm not too attached to the numbers personally.

I could probably add some automated tests for the API calls with existing data to confirm this is working consistently...upgrades across versions would be a little difficult to automate but I could at least have example 'old' volume data to boot with with whatever version I have under test.

**Comment by tehniemer** [comment]
I did flush my logs, and even start from scratch with the same problem.

http:///admin/api.php?overTimeData10mins returns ````ERR_NAME_NOT_RESOLVED````

**Comment by Noxygen** [comment]
Same here. Seems to be a known issue since v3.3. 
https://github.com/pi-hole/pi-hole/issues

**Comment by rodalpho** [comment]
Same issue here, no history at all. 

The overtimedata10mins page works fine for me but shows no domains or ads, just [] after each one.

**Comment by diginc** [comment]
I thought it was a permission problem on the volumes but they look the same when you run with or without volumes.  Running without volumes and letting the data get written inside the container instead of the host seems to work for me every time.  Specifically it seems to be /etc/pihole volume that causes the problems.

The timestamp on /etc/pihole/pihole-FTL.db keeps updating so I think it is writing the data to it but maybe php isn't pulling it back out?  That's all I could find out last night.

**Comment by oscarnyl** [comment]
I'm having the same issue. This is the output of my /admin/api.php?overTimeData10mins:

```
{
   "domains_over_time":{
      "1519797300":17,
      "1519797900":7,
      "1519798500":57,
      "1519799100":240,
      "1519799700":115,
      "1519800300":113,
      "1519800900":122,
      "1519801500":23,
      "1519802100":0,
      "1519802700":0,
      "1519803300":0,
      "1519803900":0,
      "1519804500":0,
      "1519805100":0,
      "1519805700":0,
      "1519806300":0,
      "1519806900":0,
      "1519807500":0,
      "1519808100":0,
      "1519808700":0,
      "1519809300":0,
      "1519809900":0,
      "1519810500":0,
      "1519811100":0,
      "1519811700":0,
      "1519812300":0,
      "1519812900":0,
      "1519813500":0,
      "1519814100":0,
      "1519814700":0,
      "1519815300":0,
      "1519815900":0,
      "1519816500":0,
      "1519817100":0,
      "1519817700":0,
      "1519818300":0,
      "1519818900":0,
      "1519819500":0,
      "1519820100":0,
      "1519820700":0,
      "1519821300":0,
      "1519821900":0,
      "1519822500":0,
      "1519823100":0,
      "1519823700":0,
      "1519824300":0,
      "1519824900":0,
      "1519825500":0,
      "1519826100":0,
      "1519826700":0,
      "1519827300":0,
      "1519827900":0,
      "1519828500":0,
      "1519829100":0,
      "1519829700":0,
      "1519830300":0,
      "1519830900":0,
      "1519831500":0,
      "1519832100":0,
      "1519832700":0,
      "1519833300":0,
      "1519833900":0,
      "1519834500":0,
      "1519835100":0,
      "1519835700":0,
      "1519836300":0,
      "1519836900":0,
      "1519837500":0,
      "1519838100":0,
      "1519838700":0,
      "1519839300":0,
      "1519839900":0,
      "1519840500":0,
      "1519841100":0,
      "1519841700":0,
      "1519842300":0,
      "1519842900":0,
      "1519843500":0,
      "1519844100":0,
      "1519844700":0,
      "1519845300":0,
      "1519845900":0,
      "1519846500":0,
      "1519847100":0,
      "1519847700":0,
      "1519848300":0,
      "1519848900":0,
      "1519849500":0,
      "1519850100":0,
      "1519850700":0,
      "1519851300":0,
      "1519851900":0,
      "1519852500":0,
      "1519853100":0,
      "1519853700":0,
      "1519854300":0,
      "1519854900":0,
      "1519855500":0,
      "1519856100":0,
      "1519856700":0,
      "1519857300":0,
      "1519857900":0,
      "1519858500":0,
      "1519859100":0,
      "1519859700":0,
      "1519860300":0,
      "1519860900":0,
      "1519861500":0,
      "1519862100":0,
      "1519862700":0,
      "1519863300":0,
      "1519863900":0,
      "1519864500":0,
      "1519865100":0,
      "1519865700":0,
      "1519866300":0,
      "1519866900":0,
      "1519867500":0,
      "1519868100":0,
      "1519868700":0,
      "1519869300":0,
      "1519869900":0,
      "1519870500":0,
      "1519871100":0,
      "1519871700":0,
      "1519872300":0,
      "1519872900":0,
      "1519873500":0,
      "1519874100":0,
      "1519874700":0,
      "1519875300":0,
      "1519875900":0,
      "1519876500":0,
      "1519877100":0,
      "1519877700":0,
      "1519878300":0,
      "1519878900":0,
      "1519879500":0,
      "1519880100":0,
      "1519880700":0,
      "1519881300":0,
      "1519881900":0,
      "1519882500":0,
      "1519883100":0,
      "1519883700":0,
      "1519884300":0,
      "1519884900":0,
      "1519885500":0,
      "1519886100":0,
      "1519886700":0,
      "1519887300":0,
      "1519887900":0
   },
   "ads_over_time":{
      "1519797300":1,
      "1519797900":1,
      "1519798500":6,
      "1519799100":67,
      "1519799700":37,
      "1519800300":9,
      "1519800900":5,
      "1519801500":2,
      "1519802100":0,
      "1519802700":0,
      "1519803300":0,
      "1519803900":0,
      "1519804500":0,
      "1519805100":0,
      "1519805700":0,
      "1519806300":0,
      "1519806900":0,
      "1519807500":0,
      "1519808100":0,
      "1519808700":0,
      "1519809300":0,
      "1519809900":0,
      "1519810500":0,
      "1519811100":0,
      "1519811700":0,
      "1519812300":0,
      "1519812900":0,
      "1519813500":0,
      "1519814100":0,
      "1519814700":0,
      "1519815300":0,
      "1519815900":0,
      "1519816500":0,
      "1519817100":0,
      "1519817700":0,
      "1519818300":0,
      "1519818900":0,
      "1519819500":0,
      "1519820100":0,
      "1519820700":0,
      "1519821300":0,
      "1519821900":0,
      "1519822500":0,
      "1519823100":0,
      "1519823700":0,
      "1519824300":0,
      "1519824900":0,
      "1519825500":0,
      "1519826100":0,
      "1519826700":0,
      "1519827300":0,
      "1519827900":0,
      "1519828500":0,
      "1519829100":0,
      "1519829700":0,
      "1519830300":0,
      "1519830900":0,
      "1519831500":0,
      "1519832100":0,
      "1519832700":0,
      "1519833300":0,
      "1519833900":0,
      "1519834500":0,
      "1519835100":0,
      "1519835700":0,
      "1519836300":0,
      "1519836900":0,
      "1519837500":0,
      "1519838100":0,
      "1519838700":0,
      "1519839300":0,
      "1519839900":0,
      "1519840500":0,
      "1519841100":0,
      "1519841700":0,
      "1519842300":0,
      "1519842900":0,
      "1519843500":0,
      "1519844100":0,
      "1519844700":0,
      "1519845300":0,
      "1519845900":0,
      "1519846500":0,
      "1519847100":0,
      "1519847700":0,
      "1519848300":0,
      "1519848900":0,
      "1519849500":0,
      "1519850100":0,
      "1519850700":0,
      "1519851300":0,
      "1519851900":0,
      "1519852500":0,
      "1519853100":0,
      "1519853700":0,
      "1519854300":0,
      "1519854900":0,
      "1519855500":0,
      "1519856100":0,
      "1519856700":0,
      "1519857300":0,
      "1519857900":0,
      "1519858500":0,
      "1519859100":0,
      "1519859700":0,
      "1519860300":0,
      "1519860900":0,
      "1519861500":0,
      "1519862100":0,
      "1519862700":0,
      "1519863300":0,
      "1519863900":0,
      "1519864500":0,
      "1519865100":0,
      "1519865700":0,
      "1519866300":0,
      "1519866900":0,
      "1519867500":0,
      "1519868100":0,
      "1519868700":0,
      "1519869300":0,
      "1519869900":0,
      "1519870500":0,
      "1519871100":0,
      "1519871700":0,
      "1519872300":0,
      "1519872900":0,
      "1519873500":0,
      "1519874100":0,
      "1519874700":0,
      "1519875300":0,
      "1519875900":0,
      "1519876500":0,
      "1519877100":0,
      "1519877700":0,
      "1519878300":0,
      "1519878900":0,
      "1519879500":0,
      "1519880100":0,
      "1519880700":0,
      "1519881300":0,
      "1519881900":0,
      "1519882500":0,
      "1519883100":0,
      "1519883700":0,
      "1519884300":0,
      "1519884900":0,
      "1519885500":0,
      "1519886100":0,
      "1519886700":0,
      "1519887300":0,
      "1519887900":0
   }
}
```

Note that the entries with time is before I updated the image to the most recent one.

**Comment by rroller** [comment]
I’m having the same issue where suddenly no graphs or client information. Otherwise, things are working fine. I can see DNS resolving in the logs.

Pi-hole Version v3.3 Web Interface Version v3.3 FTL Version v3.0
Running in Docker.

Your debug token is: zcoi2i3v2v

**Comment by valentingc** [comment]
Exactly the same problem since the last update.
Running Pi-Hole Version v3.3, Web Interface v3.3 and FTL v3.0.

The API call `/admin/api.php?overTimeData10mins` returns:  `{"domains_over_time":[],"ads_over_time":[]}`

DNS blocking however seems to work normally.
Also, the timestamp for `/etc/pihole/pihole-FTL.db` is updating normally.

**Comment by diginc** [comment]
Pretty sure the root cause is that the old dnsmasq config is incompatible with 3.3.  To fix this:

1. Delete /etc/dnsmasq.d/01-pihole.conf
2. Empty your /var/log/pihole.log too
  - FTL will still attempt to parse the old log lines but it wont' ever work so you'll be getting silent errors
  - I used the word 'empty' not remove because of the docker volume file mount issue described in the readme
  - example: `> /my_volumes/pihole.log` or `echo | sudo tee /var/log/pihole.log` if owned by root
3.  Restart your container to let it re-seed the 01-pihole.conf automatically

If you want to confirm /etc/dnsmasq.d/01-pihole.conf should look like sort of like this copy: https://github.com/pi-hole/pi-hole/blob/master/advanced/01-pihole.conf
The main problem for most is the lack of the `log-queries=extra` line which write dnsmasq pihole logs in the new format FTL expects.

### Or

If that doesn't work double check your timezone is correct and matches your host!  I saw some weirdness from my wrong time zone (TZ environment variable) while testing stuff out too.  I used to have `TZ=CST` and moved to `TZ=America/Chicago` style instead and that fixed what I think was a 2nd problem.

### Lastly

If none of that works I just pushed the change I had to make to see the FTL error inside the container, the ability to switch on debug mode.  This new env var is on the dev tag.  Just use dev and set the environment variable 'FTL_CMD=debug'.  If you're not debugging don't set this unless you love extra logs.

**Comment by tehniemer** [comment]
Adding ````log-queries=extra```` and re-creating the container seems to have fixed it for me. 

Is there a .conf file that contains the TZ variable? I'm not setting these containers up using the command line on my NAS and I can't find where to set my timezone.

**Comment by diginc** [comment]
TZ is just a regular docker env var passed to the docker run.  Works on all
containers AFAIK, not just mine.

**Comment by middleagedman** [comment]
It was the log-queries=extra needing to be in the dnsmasq.conf.. That's something they added in 3.3 and require.

**Comment by rodalpho** [comment]
Yes, that fixed it.

It's important to note that being extra-careful and backing up your 01-dnsmasq.conf file to, say, 01-dnsmasq.conf.BAK is a really bad idea, as dnsmasq tries to parse all files in that directory and it can be really annoying to find out why it's failing to start inside a docker container! So, like, learn from my mistakes!

**Comment by rroller** [comment]
Confirmed. Fixed for me as well

**Comment by valentingc** [comment]
Confirmed on my end aswell. Thank you!

**Comment by govardha** [comment]
I added log-queries=extra to my /etc/dnsmasq.d/01-pihole.conf and recreated/restarted my container but still the dashboard is still blank.  

Am I missing something?

**Comment by ingus16** [comment]
Confirmed on my end . Thank You so much !

**Comment by tedder** [comment]
For posterity, my 01-pihole.conf had four lines- two DNS servers, an interface, and a list. pihole.log was empty.

I added log-queries=extra, restarted container. No change.

I replaced 01-pihole.conf with [the 'advanced conf' from the pihole repo](https://github.com/pi-hole/pi-hole/blob/master/advanced/01-pihole.conf) that has 15 entries, edited my three values back in, restarted container. It works now. So whatever generates that file is wrong, apparently. I assume it'll be fixed in the next upstream release?

**Comment by mikegleasonjr** [comment]
Hi still have the same problem. It all started when I clicked "flush logs" in the admin. I used the script for installation on a Raspberry Pi.

Things I tried:  I removed `/var/log/pihole.log`, restarted `dnsmasq`.

I saw this...

==> /var/log/pihole-FTL.log <==
[2018-04-25 13:29:12.666] Warning: Reading of log file /var/log/pihole.log failed

```
root@pi3:~# ls -l /var/log/pihole.log
-rw-r----- 1 dnsmasq root 13895 Apr 25 13:30 /var/log/pihole.log
```

I changed the permissions to allow reading from all...

```
chmod 644 /var/log/pihole.log
```

I don't see the error message but still I don't see anything in the admin.
